### PR TITLE
Create UI to display resource permissions in auth profile.

### PR DESCRIPTION
**Pull request by simozhan**
JIRA: PDC-1381

### CODE REVIEW

File: pdc/apps/auth/views.py [hunk 1]
```diff
@@ -73,8 +73,41 @@ def logout(request):
...
+    dict = {}
+    result = []
+
+    if user.is_superuser:
+        resource_permission_set = set([obj for obj in models.ResourcePermission.objects.all()])
+    else:
+        group_id_list = [group.id for group in user.groups.all()]
+        queryset = models.GroupResourcePermission.objects.filter(group__id__in=group_id_list)
+        for group_resource_permission in queryset:
+            resource_permission_set.add(group_resource_permission.resource_permission)
```
Comment (ycheng-aa) [inline pdc/apps/auth/views.py]: Above part from `if`, looks almost seem as the logic in `current-user` API. Could we extract the logic to a method and reuse it in both places to eliminate duplicate code. Refactor our code a little.

**Comment by ycheng-aa** [comment]
Generally looks good. It would be nice if refactor it a little to remove duplicate.

**Comment by bliuredhat** [comment]
looks good for me

File: pdc/apps/auth/views.py (hunk 1 above)
Comment (simozhan) [inline pdc/apps/auth/views.py]: Thanks. Fixed.

File: pdc/apps/auth/views.py [hunk 2]
```diff
@@ -73,8 +73,41 @@ def logout(request):
     return HttpResponseRedirect(redirect_to)
 
 
+def _get_resource_permissions_matrix(user):
+    http_method = ['create', 'read', 'update', 'delete']
+    resource_permission_set = set([])
+    dict = {}
```
Comment (lubomir) [inline pdc/apps/auth/views.py]: I would like to see this variable renamed to not shadow a Python builtin. It will stop confusing syntax highlighters.

**Comment by lubomir** [comment]
Generally it looks good to me. One suggestion I have (and I don't know how difficult it would be to implement) is to show different text in cells where no call is possible. For example in *composes* + *create* everyone will see *no*, but there is no such call. Maybe having an empty cell or `N/A` or something like that would make it easier to understand.

**Comment by simozhan** [comment]
@lubomir yes. this is quite reasonable. fixed.

**Comment by bliuredhat** [comment]
:+1:

File: pdc/apps/auth/views.py [hunk 3]
```diff
@@ -73,8 +75,58 @@ def logout(request):
...
+        try:
+            index = http_method.index(v.lower())
+            row[index + 1] = value
+        except ValueError:
+            raise Exception("The action %s is not valid" % v)
+    return row
+
+
+def _get_resource_permissions_matrix(user):
+    d = {}
```
Comment (ycheng-aa) [inline pdc/apps/auth/views.py]: Could name `d` a more meaningful name? It's like a dict map resource name to the resource's permission list. A meaningful name looks make code easier to understand.

**Comment by ycheng-aa** [comment]
Looks good.

File: pdc/apps/auth/views.py (hunk 3 above)
Comment (simozhan) [inline pdc/apps/auth/views.py]: OK. fixed

**Comment by bliuredhat** [comment]
:+1:

**Comment by ycheng-aa** [comment]
:+1:

**Comment by lubomir** [comment]
I like it, no more comments from me.

**Pull request merged**